src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Code.java

```diff
@@ -1357,24 +1361,25 @@ void emitStackMapFrame(int pc, int localsSize) {
         }
 
         Set<VarSymbol> unsetFieldsAtPC = cpToUnsetFieldsMap.get(pc);
-        boolean generateAssertUnsetFieldsEntry = unsetFieldsAtPC != null && generateAssertUnsetFieldsFrame && !lastFrame.unsetFields.equals(unsetFieldsAtPC) ;
+        boolean generateAssertUnsetFieldsEntry = unsetFieldsAtPC != null && generateEarlyLarvalFrame && hasUninitalizedThis
```

Comment:
we should probably rename this variable

Reply:
Done, now called `encloseWithEarlyLarvalFrame`